pczt/src/common.rs

```diff
@@ -0,0 +1,202 @@
+use std::collections::BTreeMap;
+
+use getset::Getters;
+use serde::{Deserialize, Serialize};
+
+use crate::roles::combiner::merge_map;
+
+pub(crate) const FLAG_INPUTS_MODIFIABLE: u8 = 0b0000_0001;
+pub(crate) const FLAG_OUTPUTS_MODIFIABLE: u8 = 0b0000_0010;
+pub(crate) const FLAG_HAS_SIGHASH_SINGLE: u8 = 0b0000_0100;
+
```

Comment:
Shielded inputs and outputs are always signed; that [isn't dependent on the sighash flags or whether we're signing a transparent input](https://zips.z.cash/zip-0244#s-3-sapling-digest). Therefore I think we need a separate `FLAG_SHIELDED_MODIFIABLE` that is *always* set when we sign any input.